src/do-not-disturb.js

```diff
@@ -0,0 +1,53 @@
+
+// Always false if the platform doesn't support it.
+global.isDoNotDisturb = false;
+
+const LINUX_SETTING_COMMAND = "gsettings get org.gnome.desktop.notifications show-banners";
+
+function init() {
+    if (process.platform === "linux") {
+        return initForLinux();
+    }
+}
+
+async function initForLinux() {
+    // TODO: This is specific to the GNOME desktop implementation of DND
+
+    const DBus = require('dbus-next');
+    const { execSync } = require('child_process');
+    // First we need to determine the value of dnd.
```

Comment:
or second? doing it first seems racy since the value could change between us reading it and us setting up the listener (although I suspect it'll always be racy if the dbus module lacks support for inspecting the value, but maybe less so).

Reply:
>doing it first seems racy since the value could change between us reading it and us setting up the listener

Maybe, but it's a few ms at the start of the app so it feels like we can afford to be a bit tolerant of it. Alternatively we could call this each time the value changes?